src/parse.rs

```diff
@@ -1195,16 +1203,35 @@ pub(crate) struct AlignmentIndex(usize);
 
 #[derive(Clone)]
 pub(crate) struct Allocations<'a> {
-    pub refdefs: HashMap<LinkLabel<'a>, LinkDef<'a>>,
+    pub refdefs: RefDefs<'a>,
     links: Vec<(LinkType, CowStr<'a>, CowStr<'a>)>,
     cows: Vec<CowStr<'a>>,
     alignments: Vec<Vec<Alignment>>,
 }
 
+/// Keeps track of the reference definitions defined in the document.
+#[derive(Clone)]
+pub struct RefDefs<'input>(pub(crate) HashMap<LinkLabel<'input>, LinkDef<'input>>);
+
+impl<'input, 'b, 's> RefDefs<'input>
+where
+    's: 'b,
+{
+    /// Performs a lookup on reference label using unicode case folding.
+    pub fn get(&'s self, key: &'b str) -> Option<&'b LinkDef<'input>> {
+        self.0.get(&UniCase::new(key.into()))
+    }
+
+    /// Provides an iterator over all the document's reference definitions.
+    pub fn iter(&'s self) -> impl Iterator<Item = (&'s str, &'s LinkDef<'input>)> {
+        self.0.iter().map(|(k, v)| (k.as_ref(), v))
+    }
+}
+
 impl<'a> Allocations<'a> {
     pub fn new() -> Self {
         Self {
-            refdefs: HashMap::new(),
+            refdefs: RefDefs(HashMap::new()),
```

Comment:
Generally I prefer to `#[derive(Default)]` on the type and use `default` here, but that's a style point.

Reply:
Yeah, that's cleaner. I'll make the change.